gradle/libs.versions.toml

```diff
@@ -80,7 +80,7 @@ smithy-smoke-test-traits = { module = "software.amazon.smithy:smithy-smoke-test-
 junit-jupiter = { module = "org.junit.jupiter:junit-jupiter", version.ref = "junit-version" }
 junit-jupiter-params = { module = "org.junit.jupiter:junit-jupiter-params", version.ref = "junit-version" }
 
-kotlin-compile-testing = {module = "com.github.tschuchortdev:kotlin-compile-testing", version.ref = "kotlin-compile-testing-version" }
+kotlin-compile-testing = {module = "dev.zacsweers.kctfork:core", version.ref = "kotlin-compile-testing-version" }
```

Comment:
Question: Why are we changing this dependency?

Reply:
We're frequently blocked on `com.github.tschuchortdev:kotlin-compile-testing` because its Kotlin version is not upgraded often enough, leading to mismatched Kotlin compiler errors.

https://github.com/tschuchortdev/kotlin-compile-testing/issues/411

[dev.zacsweers.kctfork:core](https://github.com/ZacSweers/kotlin-compile-testing) is a fork of the project that's updated more frequently.

We can either take this fork or disable the tests.